packages/browser/core/src/core/auth/internal/StitchUserImpl.ts

```diff
@@ -19,25 +19,58 @@ import {
   StitchCredential,
   StitchUserProfileImpl
 } from "mongodb-stitch-core-sdk";
-import StitchUser from "../StitchUser";
-import StitchAuthImpl from "./StitchAuthImpl";
+import StitchRedirectCredential from "../providers/StitchRedirectCredential";
+import StitchUser, { StitchUserCodec } from "../StitchUser";
+import { prepareRedirect, RedirectFragmentFields, RedirectTypes, StitchAuthImpl } from "./StitchAuthImpl";
 
 export default class StitchUserImpl extends CoreStitchUserImpl
   implements StitchUser {
-  private readonly auth: StitchAuthImpl;
 
   public constructor(
     id: string,
     loggedInProviderType: string,
     loggedInProviderName: string,
     profile: StitchUserProfileImpl,
-    auth: StitchAuthImpl
+    private readonly auth: StitchAuthImpl
   ) {
     super(id, loggedInProviderType, loggedInProviderName, profile);
-    this.auth = auth;
   }
 
   public linkWithCredential(credential: StitchCredential): Promise<StitchUser> {
     return this.auth.linkWithCredential(this, credential);
   }
+
+  public linkUserWithRedirect(
+    credential: StitchRedirectCredential
+  ): Promise<void> {
+    this.auth.setRedirectFragment(
+      RedirectFragmentFields.LinkUser,
```

Comment:
I don't get why this is being set. This should eliminate the need for the StitchUserCodec and all changes made in in StitchUser.ts

Reply:
👍 unset